05/app.js

```diff
@@ -0,0 +1,41 @@
+function Student(firstName, lastName) {
+	this.firstName = firstName;
+	this.lastName = lastName;
+	this.grades = {};
+}
+Student.prototype.addGrade = function (subject, grade) {
+	if (!this.grades[subject]) {
+		this.grades[subject] = [];
+	}
+	this.grades[subject].push(grade);
+};
+Student.prototype.getAverageGrade = function (subject) {
+	let avgGradesArr = [];
+
+	// dla konkretnego przedmiotu
+	if (this.grades[subject] && this.grades[subject].length !== 0) {
+		const count = this.grades[subject].reduce((acc, curr) => acc + curr) / this.grades[subject].length;
+		return count;
+	}
+
+	// dla ogolnej sredniej
+	if (!subject) {
+		for (const subjectName in this.grades) {
+			const grades = this.grades[subjectName];
+			const avgGrades = grades.reduce((acc, curr) => acc + curr) / grades.length;
+			avgGradesArr.push(avgGrades);
```

Comment:
Ogólnie robienie średniej, ze średniej rodzi problemy z zaokrągleniem.
Ponieważ komputery działają w systemie dwójkowym to mamy pewne niedokładności np. 0.1+0.2=0.30000000000000004. To powoduje, że lepiej jest zsumować całość i podzielić niż operować na wartościach wymiernych. To zazwyczaj zmniejsza dokładność - to taka informacja na przyszłość :)